src/java/com/wolfssl/provider/jsse/WolfSSLEngineHelper.java

```diff
@@ -1531,6 +1538,46 @@ private boolean isLegacyDHEnabled() {
         return "legacy".equals(dhKeySize);
     }
 
+    protected synchronized boolean matchSNI(){
```

Comment:
Could you add a comment above this method, explaining what it does and if/how it applies when called on client and/or server side?